core/src/main/java/org/apache/calcite/rel/rules/SubQueryRemoveRule.java

```diff
@@ -866,6 +866,8 @@ private static void matchFilter(SubQueryRemoveRule rule,
           LogicVisitor.find(RelOptUtil.Logic.TRUE, ImmutableList.of(c), e);
       final Set<CorrelationId>  variablesSet =
           RelOptUtil.getVariablesUsed(e.rel);
+      /* Only consider the correlated variables which originated from this sub-query level*/
+      variablesSet.retainAll(filter.getVariablesSet());
```

Comment:
Interestingly, when I apply this fix in my local environment and run CoreQuidemTest, I get a regression in `misc.iq` :
```
< +-----+---+
< | KEY | C |
< +-----+---+
< |   2 | 1 |
< +-----+---+
< (1 row)
< 
---
> java.lang.AssertionError: Correlation variable $cor0 should be defined
> 	at org.apache.calcite.adapter.enumerable.EnumerableRelImplementor.getCorrelVariableGetter(EnumerableRelImplementor.java:476)
> 	at org.apache.calcite.adapter.enumerable.RexToLixTranslator.visitFieldAccess(RexToLixTranslator.java:1393)
> 	at org.apache.calcite.adapter.enumerable.RexToLixTranslator.visitFieldAccess(RexToLixTranslator.java:101)
> 	at org.apache.calcite.rex.RexFieldAccess.accept(RexFieldAccess.java:94)
...
```
Not sure why this is not shown on the PR checks...

Reply:
Yep, sorry, my bad. I confirm if I apply the fix in both `SubQueryRemoveRule` *and* `SqlToRelConverter`, then `misc.iq` test run successfully.